### PR TITLE
feature - move venue when swipe collectionView Cell

### PR DESCRIPTION
**Pull request by at-phutran**
- Change Marker when scroll collectionView Cell
- Change collectionView Cell when touch marker
- Add button next and privious of collectionView Cell

### CODE REVIEW

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: ATFoody/View/Controller/Home/LocationViewController.swift [hunk 1]
```diff
@@ -124,7 +168,10 @@ extension LocationViewController: GMSMapViewDelegate {
     func mapView(_ mapView: GMSMapView, markerInfoWindow marker: GMSMarker) -> UIView? {
         guard let infoWindow = Bundle.main.loadNibNamed(String(describing: CustomInforWindow.self), owner: nil,
             options: nil)?.first as? CustomInforWindow else { return nil }
-        infoWindow.viewModel = viewModel.infoWindowViewModel(with: marker.position)
+        infoWindow.viewModel = viewModel.infoWindowViewModel(with: marker)
+        if let indexRow = viewModel.indexOfTrending(with: marker) {
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Home/LocationViewController.swift]: check `index` in `markers` array.

File: ATFoody/View/Controller/Home/LocationViewModel.swift [hunk 2]
```diff
@@ -44,10 +46,29 @@ final class LocationViewModel: ViewModel {
         let ll = "\(lat),\(long)"
         deleteAllTrending()
         let params = Api.Venue.TrendingParams(section: "trending", offset: 0, limit: 10, venuePhotos: 1, ll: ll)
-        Api.Venue().getTrendingVenues(params: params, completion: completion)
+        Api.Venue().getTrendingVenues(params: params) { [weak self] (result) in
+            guard let this = self else { return }
+            switch result {
+            case .success(let value):
+                this.markers = []
+                for item in this.trendings {
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Home/LocationViewModel.swift]: Create a new function for init `markers`

File: ATFoody/View/Controller/Home/LocationViewModel.swift [hunk 3]
```diff
@@ -71,6 +92,14 @@ extension LocationViewModel {
         return trendings[index]
     }
 
+    func numberOfMarkers() -> Int {
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Home/LocationViewModel.swift]: `numberOfMarkers` and `numberOfTrendings` is the same value. I think we can using `numberOfItems` to replace for them

File: ATFoody/View/Controller/Home/LocationViewController.swift [hunk 4]
```diff
@@ -102,17 +105,61 @@ final class LocationViewController: UIViewController {
...
+    fileprivate func indexCollection() -> Int {
+        let contentOffsetX = venueCollectionView.contentOffset.x
+        let contentWidth = venueCollectionView.bounds.width
+        let index = Int(contentOffsetX / contentWidth)
+        return index
+    }
+
+    fileprivate func configureChangeCellButton(index: Int) {
+        previousButton.isHidden = index == 0
+        if index == viewModel.numberOfItems() - 1 {
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Home/LocationViewController.swift]: More simpler with `nextButton.isHidden = viewModel.numberOfItems() - 1`

File: ATFoody/View/Controller/Home/LocationViewController.swift [hunk 5]
```diff
@@ -102,17 +105,61 @@ final class LocationViewController: UIViewController {
...
+            nextButton.isHidden = true
+        } else {
+            nextButton.isHidden = false
+        }
+    }
+
+    fileprivate func scrollToCellAtIndex(index: Int) {
+        let indexPath: IndexPath = IndexPath(row: index, section: 0)
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Home/LocationViewController.swift]: You can no need to define `type` here.

File: ATFoody/View/Controller/Home/LocationViewController.swift [hunk 6]
```diff
@@ -102,17 +105,61 @@ final class LocationViewController: UIViewController {
...
+        selectedMarker = viewModel.marker(at: index)
+        venueCollectionView.allowsSelection = false
+    }
+}
+
+// MARK: - UIScrollViewDelegate
+extension LocationViewController: UIScrollViewDelegate {
+
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        configureChangeCellButton(index: indexCollection())
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Home/LocationViewController.swift]: Should create a new variable for returned value of `indexCollection()`

File: ATFoody/View/Controller/Home/LocationViewController.swift [hunk 7]
```diff
@@ -102,17 +105,61 @@ final class LocationViewController: UIViewController {
...
+    }
+
+    func scrollViewDidEndDecelerating(_ scrollView: UIScrollView) {
+        selectedMarker = viewModel.marker(at: indexCollection())
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Home/LocationViewController.swift]: Should create a new variable for returned value of `indexCollection()`

File: ATFoody/View/Controller/Home/LocationViewModel.swift [hunk 8]
```diff
@@ -32,6 +34,18 @@ final class LocationViewModel: ViewModel {
...
+    func createMarkers() {
+        markers.removeAll()
+        for item in trendings {
+            let latitude = item.latitude
+            let longtitude = item.longitude
+            let marker = GMSMarker()
+            marker.position = CLLocationCoordinate2D(latitude: latitude, longitude: longtitude)
+            marker.title = item.name
+            marker.icon = GMSMarker.markerImage(with: .red)
+            markers.append(marker) }
```
Comment (at-quangphu) [inline ATFoody/View/Controller/Home/LocationViewModel.swift]: The braces should be placed in a new line

File: ATFoody/View/Controller/Home/LocationViewController.swift (hunk 4 above)
Comment (at-phutran) [reply ATFoody/View/Controller/Home/LocationViewController.swift]: I would  to use the old way

**Pull request merged**